src/pages/home/ReportScreen.tsx

```diff
@@ -160,6 +165,7 @@ function ReportScreen({route, currentReportID = '', navigation}: ReportScreenPro
         }
 
         Log.info(`[ReportScreen] no reportID found in params, setting it to lastAccessedReportID: ${lastAccessedReportID}`);
+        lastAccessedReportIDRef.current = lastAccessedReportID;
         navigation.setParams({reportID: lastAccessedReportID});
```

Comment:
While the param is updating in the route, this `useEffect` can be triggered again, and `navigation.setParams` is called again with the outdated state that can cause the navigation stack to be overridden to the old state.